### DISCUSSION TITLE
prometheus scaler always tries default namespace when using dns for service from different namespace

### DISCUSSION THREAD

**Discussion by alegmal** [kedacore/keda#1967] [Q&A / Need Help] [answered]
Hi,

I feel like there is a small detail i'm missing to make it work but I'm out of ideas.
dns I tried: prometheus.monitoring.svc.cluster.local:<port> (with and without http protocol)

Error:
the HPA was unable to compute the replica count: unable to get external metric **_default_**/prometheus-prometheus-monitoring-svc-cluster-local-8080-access_frequency/&LabelSelector{MatchLabels:map[string]string{scaledObjectName: xxxx,},MatchExpressions:[]LabelSelectorRequirement{},}: unable to fetch metrics from external metrics API: the server is currently unable to handle the request (get prometheus-prometheus-monitoring-svc-cluster-local-8080-access_frequency.external.metrics.k8s.io)

I tried curl / telnet to prometheus.monitoring.svc.cluster.local from the same namespace sclaedobject is in and it works so I know networking is fine.

Keda version: 2.2.0

Any clues? Maybe I'm reading the error wrong and the problem is elsewhere?

**Comment by alegmal** [comment, marked as answer]
Found the problem to be wrongly configured firewall for GKE's private cluster.

More info to resolve the issue can be found here:
https://cloud.google.com/kubernetes-engine/docs/how-to/private-clusters#add_firewall_rules

The same guide can be accessed from KEDA's troubleshooting page:
https://keda.sh/docs/2.3/troubleshooting/

**Reply by zroubalik** [reply]
Glad you solved the problem!
